File: task-03/sub-task02/Javascript/readandwrite.js
const fs = require('fs');

const inputFilePath = 'input.txt';
const outFilePath = 'output.txt';
// reading function
fs.readFile(inputFilePath, 'utf8', (err, data) => {
    if (err) {
        console.error('Input file not found.');
        console.error(err);
        return;
    }

    fs.writeFile(outputFilePath, data, (err) => {
        if (err) {
            console.error('Error writing to output file.');
            console.error(err);
            return;
        }

        // Reading output
        fs.readFile(outputFilePath, 'utf8', (err, outputData) => {
            if (err) {
                console.error('Error reading from output file.');
                console.error(err);
                return;
            }

            console.log('Contents of output.txt:');
            console.log(outputData);
        });
    });
});

File: task-03/sub-task02/Go/readandwrite.go
package main

import (
	"fmt"
	"io/ioutil"
)

func main() {
	// Read from input file
	inputContent, err := ioutil.ReadFile("input.txt")
	if err != nil {
		fmt.Println("Error reading input file:", err)
		return
	}

	// Create and write to output file
	err = ioutil.WriteFile("output.txt", inputContent, 0644)
	if err != nil {
		fmt.Println("Error writing to output file:", err)
		return
	}

	// Read from output file to verify
	outputContent, err := ioutil.ReadFile("output.txt")
	if err != nil {
		fmt.Println("Error reading output file:", err)
		return
	}

	// Display the contents of the output file
	fmt.Println("Contents of output.txt:")
	fmt.Println(string(outputContent))
}
